rust/lance-index/src/scalar/bkd.rs

```diff
@@ -0,0 +1,373 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+//! BKD Tree (Block K-Dimensional Tree) implementation
+//!
+//! A BKD tree is a spatial index structure for efficiently indexing and querying
+//! multi-dimensional points. It's similar to a KD-tree but optimized for disk storage
+//! by grouping multiple points into leaf blocks.
+//!
+//! ## Algorithm
+//!
+//! Based on Lucene's BKD tree implementation:
+//! - Recursively splits points by alternating dimensions (X, Y)
+//! - Splits at median to create balanced tree
+//! - Groups points into leaves of configurable size
+//! - Stores tree structure separately from leaf data for lazy loading
+
+use arrow_array::{Array, ArrayRef, Float64Array, RecordBatch, UInt64Array};
+use arrow_array::cast::AsArray;
+use arrow_schema::{DataType, Field, Schema};
+use deepsize::DeepSizeOf;
+use lance_core::{Result, ROW_ID};
+use std::sync::Arc;
+
+/// BKD Tree node representing either an inner node or a leaf
+#[derive(Debug, Clone, DeepSizeOf)]
+pub struct BKDNode {
+    /// Bounding box: [min_x, min_y, max_x, max_y]
+    pub bounds: [f64; 4],
+    /// Split dimension: 0=X, 1=Y
+    pub split_dim: u8,
+    /// Split value along split_dim
+    pub split_value: f64,
+    /// Left child node index (for inner nodes)
+    pub left_child: Option<u32>,
+    /// Right child node index (for inner nodes)
+    pub right_child: Option<u32>,
+    /// Leaf ID - corresponds to leaf_{id}.lance file (for leaf nodes)
+    pub leaf_id: Option<u32>,
+}
+
+/// In-memory BKD tree structure for efficient spatial queries
+#[derive(Debug, DeepSizeOf)]
+pub struct BKDTreeLookup {
```

Comment:
nit: we prefer to still have camel case style for abbreviations, for example `Hnsw` instead of `HNSW`. So could you update `BKD` related terms to `Bkd`